pico/server.py

```diff
@@ -33,6 +36,25 @@ def run_app(app, ip='127.0.0.1', port=4242, use_debugger=True, use_reloader=True
     if len(sys.argv) > 2:
         server_ip = sys.argv[2]
     else:
-        server_ip = '127.0.0.1'
+        server_ip = DEF_SERVER_IP
+
+    if ':' in server_ip:
+        server_ip, server_port = server_ip.split(':')
+
+        # For now, werkzeug defaults to localhost if ip == '', but may change.
+        if not server_ip:
+            server_ip = DEF_SERVER_IP
+
+        try:
+            server_port = int(server_port)
+        except ValueError:
+            try:
+                server_port = socket.getservbyname(server_port, 'tcp')
+
+            # OSError in python3, but error in python2.7.
+            except Exception:
+                server_port = DEF_SERVER_PORT
```

Comment:
I don't think we need to handle named ports. It seems like quite an uncommon use case.

If we get a `ValueError` we should raise a `TypeError` to inform the user that the port part of the string was not a valid port. e.g.:
`raise TypeError('Invalid port number specified.')`